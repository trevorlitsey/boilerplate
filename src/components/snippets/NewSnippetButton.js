import React from 'react';
import { func } from 'prop-types';
import styled from 'styled-components';

const Button = styled.button`
	width: 100%;
	background: none;
	border: 1.4px solid var(--gray);
	padding: 10px;
	transition: all .2s;
	cursor: pointer;

	&:hover {
		background: var(--light);
	}
`

const iconStyles = {
	margin: '0 10px',
	color: 'var(--gray)',
}

const NewSnippetButton = ({ showModal }) => (
	<Button onClick={showModal} type="button" className="rounded">
		+ Add New Snippet
	</Button>
)

NewSnippetButton.propTypes = {
	showModal: func.isRequired,
}


export default NewSnippetButton;