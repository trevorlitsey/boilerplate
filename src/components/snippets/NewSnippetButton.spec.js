import React from 'react';
import { shallow } from 'enzyme';

import NewSnippetButton from './NewSnippetButton';

describe('NewSnippetButton', () => {

	it('should render without crashing', () => {
		const wrapper = renderNewSnippetButton();
		expect(wrapper).toMatchSnapshot();
	})

	it('should call showModal on click', () => {
		const showModal = jest.fn();
		const wrapper = renderNewSnippetButton({ showModal });
		wrapper.find('[type="button"]').simulate('click');
		expect(showModal).toHaveBeenCalled();
	})

})

// -------
function renderNewSnippetButton(props = {}) {
	const propsToUser = {
		showModal: () => { },
		...props,
	}

	return shallow(<NewSnippetButton {...propsToUser} />)
}