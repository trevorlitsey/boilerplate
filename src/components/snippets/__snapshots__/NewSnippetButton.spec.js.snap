// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewSnippetButton should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewSnippetButton
    showModal={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "+ Add New Snippet",
      "className": "rounded",
      "onClick": [Function],
      "type": "button",
    },
    "ref": null,
    "rendered": "+ Add New Snippet",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "+ Add New Snippet",
        "className": "rounded",
        "onClick": [Function],
        "type": "button",
      },
      "ref": null,
      "rendered": "+ Add New Snippet",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
